### PR TITLE
Riot: migrate to TS

### PR DESCRIPTION
**Pull request by gaetanmaisse**
Issue: Part of #5030

## What I did

Fastly migrate `@storybook/riot` to TS as I want to close the big TS Migration Issue before moving to improve types of `@storybook/core`.

## How to test

- CI should be 🟢 , we rely only on examples as there is no E2E for Riot (it looks like there are some issues with Riot 4 + SB 🤷🏻 )

### CODE REVIEW

File: app/riot/src/client/preview/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { start } from '@storybook/core/client';
 
 import './globals';
+// @ts-ignore
```
Comment (gaetanmaisse) [inline app/riot/src/client/preview/index.ts:4]: There are some miscellaneous things with `tag2` across many files, it's imported from `riot` (but isn't available is the type), sometimes it's just stored in an unused `const` but if you remove them everything breaks 🧙🏻

File: app/riot/src/client/preview/render-riot.test.ts [hunk 2]
```diff
@@ -23,18 +24,21 @@ beforeEach(() => {
 
 describe('render a riot element', () => {
   it('should not work with nothing', () => {
+    // @ts-ignore
```
Comment (gaetanmaisse) [inline app/riot/src/client/preview/render-riot.test.ts:27]: Same here... `render`  should only have 1 argument but in every test, there is a magic `context` given too.

File: app/riot/src/client/preview/render-riot.test.ts [hunk 3]
```diff
@@ -23,18 +23,22 @@ beforeEach(() => {
 
 describe('render a riot element', () => {
   it('should not work with nothing', () => {
+    // `render` should only have 1 argument but in every test, there is a magic context given too… 🤷
```
Comment (ThibaudAV) [inline app/riot/src/client/preview/render-riot.test.ts:26]: I think the 2nd argument can be deleted without any problem.

I think the test has been missed with this commit 
https://github.com/storybookjs/storybook/commit/f555ce0d6956a8fcb9cfd7d9c0fbe899de401bdc#diff-dc99c8ceca53cf99aca341cfaa97633689b0d7edbdaa26631026779cd0e723dc

**Review by ThibaudAV — APPROVED** [review]
LGTM

File: app/riot/src/client/preview/render-riot.test.ts (hunk 3 above)
Comment (gaetanmaisse) [reply app/riot/src/client/preview/render-riot.test.ts]: Thanks, you're right, I thought I tried to remove them and everything was OK but I just retried, all tests are 🟢 . 🙇🏻

**Pull request merged**